src/os/bluestore/BlueStore.cc

```diff
@@ -3661,10 +3672,26 @@ void BlueStore::ExtentMap::reshard(
     }
     dout(30) << " extent " << *extent << dendl;
 
-    // disfavor shard boundaries that span a blob
-    bool would_span = (extent->logical_offset < max_blob_end) || extent->blob_offset;
-    if (estimate &&
-	estimate + extent_avg > target + (would_span ? slop : 0)) {
+
+    bool make_shard_here = false;
+    if (onode_data_has_boundaries) {
+      if (extent->blob_start() >= next_boundary) {
+	// this it the place we want to have shard boundary
+	if ((estimate >= target /*we have enough already*/) ||
+	    (estimate + encoded_segment_estimate >= target * 3 / 2)
+	    /*we will be too large if we wait for next segment*/) {
+	  make_shard_here = true;
+	}
+	next_boundary = p2roundup(extent->blob_end(), segment_size);
+      }
+    } else {
+      // disfavor shard boundaries that span a blob
+      bool would_span = (extent->logical_offset < max_blob_end) || (extent->blob_offset != 0);
+      if (estimate && estimate + extent_avg > target + (would_span ? slop : 0)) {
```

Comment:
please please (here and earlier) - add the unneeded brackets. Sure - the compiler handles ops order correctly, but it's a burden on the reader of the code.

Reply:
I accept the logic and validity of your request here.
As I mentally inherited this code from Sage and had to work through its convoluted logic I ceased to see its minor inconveniences.
Please keep pointing to the locations you think need more "()"s.